plugins/incremental-ingestion-backend/src/routes.ts

```diff
@@ -0,0 +1,230 @@
+import { IncrementalIngestionDatabaseManager } from './database/IncrementalIngestionDatabaseManager';
+
+/** @public */
+export const createIncrementalProviderRouter = async (
+  manager: IncrementalIngestionDatabaseManager,
+  logger: Logger,
+) => {
+  const router = Router();
+  router.use(express.json());
+
+  // Get the overall health of all incremental providers
+  router.get('/health', async (_, res) => {
+    const records = await manager.healthcheck();
+    const providers = records.map(record => record.provider_name);
+    const duplicates = [
+      ...new Set(providers.filter((e, i, a) => a.indexOf(e) !== i)),
+    ];
+
+    if (duplicates.length > 0) {
+      res.json({ healthy: false, duplicateIngestions: duplicates });
+    } else {
+      res.json({ healthy: true });
+    }
+  });
+
+  // Clean up and pause all providers
+  router.delete('/cleanup', async (_, res) => {
+    const result = await manager.cleanupProviders();
+    res.json(result);
+  });
+
+  // Get basic status of the provider
+  router.get('/:provider', async (req, res) => {
+    const { provider } = req.params;
+    const record = await manager.getCurrentIngestionRecord(provider);
+    if (record) {
+      res.json({
+        success: true,
+        status: {
+          current_action: record.status,
+          next_action_at: new Date(record.next_action_at),
+        },
+        last_error: record.last_error,
+      });
+    } else {
+      const providers: string[] = await manager.listProviders();
+      if (providers.includes(provider)) {
+        res.json({
+          success: true,
+          status: {
+            current_action: 'rest complete, waiting to start',
+          },
+        });
+      } else {
+        logger.error(
+          `${provider} - No ingestion record found in the database!`,
+        );
+        res.status(404).json({
+          success: false,
+          status: {},
+          last_error: `Provider '${provider}' not found`,
+        });
+      }
+    }
+  });
+
+  // Trigger the provider's next action
+  router.put('/:provider', async (req, res) => {
+    const { provider } = req.params;
+    const record = await manager.getCurrentIngestionRecord(provider);
+    if (record) {
+      await manager.triggerNextProviderAction(provider);
+      res.json({
+        success: true,
+        message: `${provider}: Next action triggered.`,
+      });
+    } else {
+      const providers: string[] = await manager.listProviders();
+      if (providers.includes(provider)) {
+        logger.debug(`${provider} - Ingestion record found`);
+        res.json({
+          success: true,
+          message: 'Unable to trigger next action (provider is restarting)',
+        });
+      } else {
+        res.status(404).json({
+          success: false,
+          message: `Provider '${provider}' not found`,
+        });
+      }
+    }
+  });
+
+  // Start a brand-new ingestion cycle for the provider
+  router.post('/:provider', async (req, res) => {
```

Comment:
could this be `POST /providers/:provider/start` instead?

Reply:
Done!